src/providers/swagger/swagger.provider.ts

```diff
@@ -0,0 +1,256 @@
+  );
+};
+
+/**
+ * generate the Open API schema from endpoint
+ */
+const apiSchemaFromEndpoint = (
+  key: string,
+  e: MinimalEndpointInstance,
+  tags: string[]
+): any => {
+  const path = e.getStaticPath((param: any) => `:${param}`);
+  const responseStatusCode = e.Method === 'POST' ? 201 : 200;
+
+  const input = e.Input;
+
+  const Headers = (input as any).Headers;
+
+  // derive headers from io-ts definition of e.Input.Headers
+  const headerParameters =
+    Headers !== undefined
+      ? R.keys(Headers.props).reduce<any[]>(
+          (acc, k) =>
+            acc.concat({
+              name: k,
+              in: 'header',
+              required: false,
+              schema: getOpenAPISchema((Headers?.props)[k]),
+            }),
+          []
+        )
+      : [];
+
+  // add security when "x-authorization" header is defined
+  const security =
+    Headers?.props?.['x-authorization'] !== undefined ? [{ ACTToken: [] }] : [];
+
+  const Query = (input as any).Query;
+
+  // derive query parameters from io-ts definition of e.Input.Query
+  const queryParameters =
+    Query !== undefined
+      ? R.keys(Query.props).reduce<any[]>(
+          (acc, k) =>
+            acc.concat({
+              name: k,
+              in: 'query',
+              required: Query?._tag === 'PartialType',
+              schema: getOpenAPISchema((Query?.props)[k]),
+            }),
+          []
+        )
+      : [];
+
+  // console.log('endpoint query', e.Input.Params);
+
+  const Params = (input as any).Params;
+  // derive path parameters from io-ts definition of e.Input.Params
+  const pathParameters =
+    Params !== undefined
+      ? R.keys(Params.props).reduce<any[]>(
+          (acc, k) =>
+            acc.concat({
+              name: k,
+              in: 'path',
+              required: true,
+              schema: getOpenAPISchema((Params?.props)[k]),
+            }),
+          []
+        )
+      : [];
+
+  // combine all parameters
+  const parameters = [
+    ...headerParameters,
+    ...queryParameters,
+    ...pathParameters,
+  ];
+  //   console.log('parameters', parameters);
+
+  // add definition of request body, if needed
+  const requestBody = hasRequestBody(e)
+    ? {
+        content: {
+          'application/json': {
+            schema: getOpenAPISchema((input as any)?.Body),
+          },
+        },
+      }
+    : {};
+
+  // define success response
+  const successResponse = {
+    [responseStatusCode]: {
+      description: (e.Output as any).name,
+      content: {
+        'application/json': {
+          schema: getOpenAPISchema(e.Output as any),
+        },
+      },
+    },
+  };
+
+  // TODO: define error response
+
+  return {
+    summary: key,
+    description: `${e.Method} ${path}`,
+    tags: tags,
+    parameters,
+    security,
+    requestBody,
+    responses: {
+      ...successResponse,
+    },
+  };
+};
+
+export const generateDoc = (config: DocConfig): any => {
+  const paths = pipe(
```

Comment:
So basically this walks though endpoints exported from backend and generate an object with paths

Reply:
@vecna